src/lib.rs

```diff
@@ -107,25 +146,58 @@ pub fn run() -> Result<Vec<Warning>, Box<dyn Error>> {
 
     let mut warnings: Vec<Warning> = Vec::new();
     for file in new_files {
-        let f = File::open(&file.path)?;
-        let reader = BufReader::new(f);
+        let result = run_checks(&file)?;
+        warnings.extend(result);
+    }
+
+    Ok(warnings)
+}
 
-        // TODO: Initialize a vector with a capacity equal to the number of lines
-        let mut lines: Vec<LineEntry> = Vec::new();
-        for (index, line) in reader.lines().enumerate() {
-            let number = index + 1;
-            let raw_string = line?;
+#[cfg(test)]
+mod tests {
+    use super::*;
 
-            lines.push(LineEntry {
-                number,
-                file_path: file.path.clone(),
-                raw_string,
-            })
-        }
+    use std::io::Write;
+    use tempfile::{NamedTempFile, TempPath};
 
-        let result = checks::run(lines);
-        warnings.extend(result);
+    pub fn create_temp_file(contents: &str) -> TempPath {
+        let mut file = NamedTempFile::new().unwrap();
+        file.write(contents.as_bytes()).unwrap();
+
+        file.into_temp_path()
     }
 
-    Ok(warnings)
+    #[test]
+    fn correct_files() {
+        let contents = vec![
+            "A=B\nF=BAR\nFOO=BAR\n",
+            "A=B\r\nF=BAR\r\nFOO=BAR\r\n",
+            "# comment\nABC=DEF\n",
```

Comment:
I think it is more integration test, not unit. Please, move it into the `tests` directory.

Reply:
In this test I wanted to test working with a single file, independently of file browsing.

So I need to make the `run_checks` function and the `common::FileEntry` type public, to make the test an integration one.

Reply:
What do you think, @mgrachev?

Reply:
I think you do something wrong if you want to make some functions public only for tests.

Reply:
The `lib.rs` file shoud check only in intergration tests.

Reply:
> The `lib.rs` file shoud check only in intergration tests.

I agree with this, but my wish may indicate that it is needed some decomposition in `lib.rs` )

I slightly change approach and move this tests to `tests` directory.

Also I noticed that we can't check a file not from the current directory (or its subdirectories):
```
:~$ dotenv-linter 1.env
1.env:3 LowercaseKey: The fOO key should be in uppercase
:~$ cd subd
:~/subd$ dotenv-linter ~/1.env
:~/subd$
```
It is because `let strip_path = match result` in `lib.rs` (`StripPrefixError`).

Is this done on purpose?

Reply:
> > The `lib.rs` file shoud check only in intergration tests.
> 
> I agree with this, but my wish may indicate that it is needed some decomposition in `lib.rs` )

Absolutely agree. I plan to do a refactor this file a bit later and cover it with tests.

> 
> I slightly change approach and move this tests to `tests` directory.
> 
> Also I noticed that we can't check a file not from the current directory (or its subdirectories):
> 
> ```
> :~$ dotenv-linter 1.env
> 1.env:3 LowercaseKey: The fOO key should be in uppercase
> :~$ cd subd
> :~/subd$ dotenv-linter ~/1.env
> :~/subd$
> ```
> 
> It is because `let strip_path = match result` in `lib.rs` (`StripPrefixError`).
> 
> Is this done on purpose?

I know about that problem. Could you open an issue about this problem?
There we can discuss it is a real problem or not.

Reply:
https://github.com/dotenv-linter/dotenv-linter/issues/179